### DISCUSSION TITLE
Loop error on form submit since 7.45.0

### DISCUSSION THREAD

**Discussion by giovannetti-eric** [react-hook-form/react-hook-form#10568] [Q&A]
Since 7.45.0, I get a warning at form submit or at email change that make my app unresponsive:

> Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.

It seems related to the use of useForm and useWatch.

Replacing this code that was working before 7.45.0:

```
const email = searchParams.get("email");
```

```
const methods = useForm<IFormInput>({
    values: {
      email: email || "",
      password: "",
    },
    resetOptions: {
      keepDirtyValues: true,
    },
});
```

by:

```
const methods = useForm<IFormInput>();
```

Is fixing the error loop on form submit and removing this code fix the loop error on email change:

```
const emailInput = useWatch({
    control: methods.control,
    name: "email",
});
```

I have read the change of the 7.45.0, that not supposed be a breaking change update, but I'm not sure what cause that.

**Comment by leapful** [comment]
@giovannetti-eric 

Please provide a minimal codesandbox to re-produce the issue. Based on the description, the problem is from `useEffect` handling, not from `react-hook-form`. It might be related to some dependencies of `useEffect` that have side-effect with change from newer version of `react-hook-form`
